### PR TITLE
executor: add timestamp to a log

### PR DESCRIPTION
**Pull request by fzhedu**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/31012

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

[INFO] [conn.go:1115] ["command dispatched failed"] [conn=7] [connInfo="id:7, addr:127.0.0.1:53237 status:10, collation:utf8mb4_0900_ai_ci, use       r:root"] [command=Query] [status="inTxn:0, autocommit:1"] [sql="select count(*) from nation"] [txn_mode=PESSIMISTIC] [err="rpc error: code = Unavailable desc = connection error       : desc = \"transport: Error while dialing dial tcp 127.0.0.1:3930: connect:

after

["command dispatched failed"] [conn=5] [connInfo="id:5, addr:127.0.0.1:51078 status:10, collation:utf8mb4_0900_ai_ci, user:root"] [command=Query] [status="inTxn:0, autocommit:1"] [sql="select count(*) from nation"] [txn_mode=PESSIMISTIC] **[timestamp=430394964705280001]** [err="[tikv:9012]TiFlash server timeout

- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: server/conn.go [hunk 1]
```diff
@@ -1117,6 +1117,7 @@ func (cc *clientConn) Run(ctx context.Context) {
 					zap.String("status", cc.SessionStatusToString()),
 					zap.Stringer("sql", getLastStmtInConn{cc}),
 					zap.String("txn_mode", txnMode),
+					zap.Uint64("timestamp", cc.ctx.TxnInfo().StartTS),
```
Comment (cfzjywxk) [inline server/conn.go:1120]: The `ctx` could be nil if the session opeing fails, also the returned `TxnInfo` could be nil if the `txn` is not activiated.

Comment (fzhedu) [reply server/conn.go:1121]: > The `ctx` could be nil if the session opeing fails, also the returned `TxnInfo` could be nil if the `txn` is not activiated.

would  `ctx` be nil ？ if yes, then the next line also has a bug.

**Review by guo-shaoge — APPROVED** [review]

**Review by cfzjywxk — APPROVED** [review]

**Pull request merged**